barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_integration.test.cpp

```diff
@@ -322,14 +367,72 @@ TEST_P(AcirIntegrationFoldingTest, ProveAndVerifyProgramStack)
     }
 }
 
+TEST_P(AcirIntegrationFoldingTest, FoldAndVerifyProgramStack)
+{
+    using Flavor = GoblinUltraFlavor;
+    using Builder = Flavor::CircuitBuilder;
+
+    std::string test_name = GetParam();
+    auto program_stack = get_program_stack_data_from_test_file(test_name);
+
+    ClientIVC ivc;
+    ivc.structured_flag = true;
+
+    while (!program_stack.empty()) {
+        auto program = program_stack.back();
+
+        // Construct a bberg circuit from the acir representation
+        auto circuit =
+            acir_format::create_circuit<Builder>(program.constraints, 0, program.witness, false, ivc.goblin.op_queue);
+
+        ivc.accumulate(circuit);
+
+        CircuitChecker::check(circuit);
+        // EXPECT_TRUE(prove_and_verify_honk<Flavor>(ivc.prover_instance));
+
+        program_stack.pop_back();
+    }
+
+    EXPECT_TRUE(ivc.prove_and_verify());
+}
+
 INSTANTIATE_TEST_SUITE_P(AcirTests,
                          AcirIntegrationFoldingTest,
-                         testing::Values("fold_after_inlined_calls",
-                                         "fold_basic",
-                                         "fold_basic_nested_call",
-                                         "fold_call_witness_condition",
-                                         "fold_complex_outputs",
-                                         "fold_distinct_return",
-                                         "fold_fibonacci",
-                                         "fold_numeric_generic_poseidon"));
-#endif
\ No newline at end of file
+                         testing::Values("fold_basic", "fold_basic_nested_call"));
```

Comment:
Restricted this to just two tests since some of them have >100 circuits which would take forever with clientIvc, especially because we need to use the structured trace to handle the different sized circuits. All of the fold tests are already tested in the acir_tests suite anyway